Add some developer doc and clean up an error

mulby opened the pull request:
Note that run_tests.sh was reporting an error, however the CI tests were passing. I added an explicit exit command to ensure the build fails if there are any failed projects.

I also added an explicit declaration of the shell to the top of run_tests.sh since I run zsh by default, and this script relies on bash specific features.

Also added the "sphinx-autobuild" tool and a "develop.sh" script that can be used to run it. It opens up a live HTML server that reloads your documentation every time it is saved, allowing you to preview it every time you save a file.

@lamagnifica @mhoeber

lamagnifica commented on README.rst:
```diff
@@ -50,4 +50,19 @@ Flow`_.
\ No newline at end of file
+docs@edx.org.
+
+Before submitting a pull request, it is recommend you run the test suite on
```
it is recommend --> it is recommended

(I'm not sure that this actually applies to doc-only PRs, but...)

mulby replied:
Definitely not doc specific, I just thought I would make it *really* clear how one goes about running the tests. It wasn't immediately clear to me that I should be using the shared/travis_requirements.txt file to setup my environment (for example).

Might be overkill, but I did the following to help the next new developer to land here:

* creating a symlink called "requirements.txt" which is a fairly standard convention for listing third party python dependencies
* updating this readme with these instructions (even though they are arguably obvious)

lamagnifica commented on en_us/course_authors/source/cohorts/cohort_config.rst:
```diff
@@ -124,11 +124,11 @@ divided by cohort. See :ref:`Read the Cohort Indicator in Posts`.
+         "Example Group Name A",
+         "Example Group Name B",
+         "Example Group Name C"
+     ]
```
This spacing change makes me anxious. This example should look as much as possible like the screen shot that follows it in the text (click View for this file to see the image).

mulby replied:
reStructuredText doesn't recongize this as a code block without this change, so I imagine it looked extra funky without this.

That said, I'm not sure what this *should* look like, do you know who "owns" this part of the doc?

This was the one error that was being thrown by sphinx.

lamagnifica replied:
Carol owns it. The example code block is meant to mimic how the lines look when they get entered in Studio, and the Studio "save" action reformats it for you. Here's the image I mentioned above: https://github.com/mulby/edx-documentation/blob/gabe/framework-tweaks/en_us/course_authors/source/cohorts/cohort_config.rst#define-the-auto-cohort-groups
So, I'm glad to understand why you made the change. It alleviates some of my anxiety!

mulby replied:
@catong can you sanity check this change?

catong replied:
@mulby this is fine. Thanks!

lamagnifica commented:
Thanks, Gabe! I'm not sure what project(s) this relates to, and I don't have the expertise to give thumbs, so I've offered a few suggestions.

mulby commented:
@lamagnifica this change is definitely not project-specific. It looks like Christine may have set it up initially.

@clytwynec can you take a look at my changes here?

clytwynec commented:
Changes seem good to me :+1:

mulby commented:
@clytwynec do you know what the really heavyweight shared/requirements.txt is for? It almost looks like an LMS requirements file...

I'm a little nervous I'm making things more confusing by adding a symlink in the repo root.

clytwynec commented:
@mulby  I don't think that file is actually used for anything here.  I think it was just copied over from [the platform repo](https://github.com/edx/edx-platform/blob/master/docs/shared/requirements.txt).  

@mhoeber  Can you confirm that?

mulby commented:
@mhoeber this one is waiting on you when you get a chance, but it's very low priority... just adds some tools to the repo and fixes a bug in the test script.

mulby commented:
Rebased on master. Removed the merge conflict with the latest cohort doc. Looks like the error I found has been cleaned up in the latest revision of that document.

@mhoeber now this is purely a convenience patch, adds some developer doc and a helper script for people working on the docs. Let me know if it's OK to merge.

mhoeber commented:
sorry for delay, thanks, :+1:
